### PR TITLE
[Documentation] Rework for v0.2.2

### PR DESCRIPTION
**Pull request by skim0119**
# Documentation Update

Here is the sample build of the updated documentation pages. There might be a few updates/fixes that are necessary, but I think we can continue working on those in branch `update-0.2.2`.

https://docs.cosseratrods.org/en/doc_patch/index.html

## Key Changes

- The workflow and tutorial in the webpage are merged in the documentation page.
- Each module is listed in the table.
- Compatibility table is added to indicate which modules are supported in rigid body vs rod.
- Advanced guide is added to archive important discussions.
- Old information is removed, new information are added 😄

### CODE REVIEW

File: elastica/rod/cosserat_rod.py [hunk 1]
```diff
@@ -34,83 +36,82 @@ class CosseratRod(RodBase):
...
+            Rod element radius.
+        dilatation: numpy.ndarray
+            1D (blocksize) array containing data with 'float' type.
+            Rod element dilatation.
+        voronoi_dilatation: numpy.ndarray
+            1D (blocksize) array containing data with 'float' type.
+            Rod dilatation on voronoi domain.
+        dilatation_rate: numpy.ndarray
+            1D (blocksize) array containing data with 'float' type.
+            Rod element dilatation rates.
```
Comment (armantekinalp) [inline elastica/rod/cosserat_rod.py:114]: Instead of block size, can we change to `n_nodes`, `n_elems`, n_voronoi`

**Review by armantekinalp — APPROVED** [review]
After the changes we can merge this documentation. LGTM

**Comment by armantekinalp** [comment]
I also reformatted some documentation @skim0119

File: docs/guide/example_cases.rst [hunk 2]
```diff
@@ -0,0 +1,31 @@
+*************
+Example Cases
+*************
+
+Example cases are the demonstration of physical example with known analytical solution or well-studied phenomenon.
+Each cases follows the recommended workflow, shown [here](./workflow.md). Feel free to use them as an initial template to build your own case study.
```
[the remark this answers is not in the document]
Comment (skim0119) [reply docs/guide/example_cases.rst]: Fixed

File: elastica/rod/cosserat_rod.py (hunk 1 above)
Comment (skim0119) [reply elastica/rod/cosserat_rod.py]: changed

**Pull request merged**